src/attributes/limits.md

```diff
@@ -7,8 +7,42 @@ The following [attributes] affect compile-time limits.
 The *`recursion_limit` attribute* may be applied at the crate level to set the
 maximum depth for potentially infinitely-recursive compile-time operations
 like auto-dereference or macro expansion. It uses the [_MetaNameValueStr_]
-syntax to specify the recursion depth. The default is
-`#![recursion_limit="64"]`.
+syntax to specify the recursion depth. The default is 64.
+
+```rust,ignore
+#![recursion_limit = "4"]
+
+macro_rules! a {
+    () => { a!(1) };
+    (1) => { a!(2) };
+    (2) => { a!(3) };
+    (3) => { a!(4) };
+    (4) => { };
+}
+
+// This fails to expand because it requires a recursion depth greater than 4.
+a!{}
+```
+
+## The `type_length_limit` attribute
+
+The *`type_length_limit` attribute* limits the maximum size of a type
+constructed during monomorphization. It is applied at the crate level, and
```

Comment:
"crate" should link to the crate and source files page.

I think we should say "concrete type" instead of "type constructed during monomorphization" and we should add "concrete type" to the glossary/make sure its defined in one of the types chapters. I'll file an issue for that one.

Reply:
I've reworded it in a way that is maybe a little clearer. I think it is important to make the distinction that it is the instantiation of an abstract type to a concrete one that triggers this check.

(Also, this is far beyond my knowledge of the compiler, so this is the best to my understanding of reading the code and writing tests.)